ProiectPractica/Validator/ConfigurationValidator.cs

```diff
@@ -0,0 +1,133 @@
+        /// <param name="errorMessage"></param>
+        /// <returns></returns>
+        public bool ValidateFloat(string textToBeValidated, out string errorMessage)
+        {
+            if (textToBeValidated.Length == 0)
+            {
+                errorMessage = "Field should not be empty!";
+                return false;
+            }
+            if(float.TryParse(textToBeValidated, CultureInfo.CreateSpecificCulture("de-DE"), out float number))
+            {
+                if (number < 0)
+                {
+                    errorMessage = "The number should be positive!";
+                    return false;
+                }
+                if (textToBeValidated.Contains('.'))
+                {
+                    errorMessage = "The number is not in correct format!";
+                    return false;
+                }
+                if (number > float.MaxValue)
+                {
+                    errorMessage = "The number is too big";
+                    return false;
+                }
+                // after 7 decimals it will approximate the number, whitch can impact the configuration saved in the file
+                if(textToBeValidated.Contains(',') && textToBeValidated.Substring(textToBeValidated.IndexOf(',')).Length > 7)
+                {
+                    errorMessage = "Too many decimals!";
+                    return false;
+                }
+                errorMessage = "";
+                return true;
+            }
+            errorMessage = "The number is not in correct format!";
+            return false;
+        }
+
+        /// <summary>
+        /// Validates a string so that it can be parsed into an integer that represents an hour.
+        /// Returns true if the string can be parsed as an hour,
+        /// and it returns false otherwise.
+        /// It also checks if the end hour is greater than start hour
+        /// </summary>
+        /// <param name="textToBeValidated"></param>
+        /// <param name="errorMessage"></param>
+        /// <returns></returns>
+        public bool ValidateHour(string textToBeValidated, out string errorMessage)
+        {
+            if (textToBeValidated.Length == 0)
+            {
+                errorMessage = "Field should not be empty!";
+                return false;
+            }
+            if (int.TryParse(textToBeValidated, out int hour))
+            {
+                // -0 passed the validation, so this: textToBeValidated[0] == '-', assures that -0 will not pass the validation
+                if (hour < 0 || hour > 23 || textToBeValidated[0] == '-')
+                {
+                    errorMessage = "Invalid hour!";
+                    return false;
+                }
+                errorMessage = "";
+                return true;
+            }
+            else if ((!textToBeValidated.Substring(1).All(char.IsDigit) && textToBeValidated[0] != '-') || !textToBeValidated.All(char.IsDigit))
```

Comment:
Why are you checking if the number is negative after failing to parse to int?

Reply:
The same reason as above. I should have left more comments so the code would have been clearer.